### PR TITLE
[dagster-gcp] rename gcs io manager to GCSPickleIOManager

### PR DESCRIPTION
**Pull request by jamiedemaria**
## Summary & Motivation

Rename `ConfigurablePickledObjectGCSIOManager` to `GCSPickleIOManager `as per discussion in https://github.com/dagster-io/dagster-website/pull/667

### CODE REVIEW

**Comment by jamiedemaria** [comment]
Current dependencies on/for this PR:
* master
  * **PR #15894** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15894" target="_blank">Graphite</a> 
    * **PR #15895** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15895" target="_blank">Graphite</a>  👈
      * **PR #15896** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15896" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/15895?utm_source=stack-comment).

File: docs/content/concepts/io-management/io-managers.mdx [hunk 1]
```diff
@@ -821,7 +821,7 @@ Additionally, if the handled output is part of a software-defined asset, these m
...
 | <PyObject module="dagster_aws.s3" object="S3PickleIOManager" />                            | Stores outputs as pickle files in Amazon Web Services S3.                     |                                                                                                                        |
 | <PyObject module="dagster_azure.adls2" object="ConfigurablePickledObjectADLS2IOManager" /> | Stores outputs as pickle files in Azure ADLS2.                                |                                                                                                                        |
-| <PyObject module="dagster_gcp.gcs" object="ConfigurablePickledObjectGCSIOManager" />       | Stores outputs as pickle files in Google Cloud Platform GCS.                  |                                                                                                                        |
+| <PyObject module="dagster_gcp.gcs" object="GCSPickleIOManager" />                          | Stores outputs as pickle files in Google Cloud Platform GCS.                  |                                                                                                                        |
```
Comment (erinkcochran87) [inline docs/content/concepts/io-management/io-managers.mdx:824]: Same here as on the S3 PR - the PyObject link isn't working for me. Potential Sphinx thing?

File: python_modules/libraries/dagster-gcp/dagster_gcp/gcs/io_manager.py [hunk 2]
```diff
@@ -168,9 +168,13 @@ def handle_output(self, context: OutputContext, obj: Any) -> None:
         self._internal_io_manager.handle_output(context, obj)
 
 
+# GCSPickleIOManager used to be named ConfigurablePickledObjectGCSIOManager, keep this symbol around for backcompat
+ConfigurablePickledObjectGCSIOManager = GCSPickleIOManager
```
Comment (schrockn) [inline python_modules/libraries/dagster-gcp/dagster_gcp/gcs/io_manager.py:172]: similar to other diff, consider making this a deprecated subclass

**Review by schrockn — APPROVED** [review]

File: docs/content/concepts/io-management/io-managers.mdx (hunk 1 above)
Comment (erinkcochran87) [reply docs/content/concepts/io-management/io-managers.mdx]: All good here!

**Pull request merged**